packages/@aws-cdk/aws-scheduler-targets-alpha/lib/universal.ts

```diff
@@ -0,0 +1,110 @@
+import { IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { awsSdkToIamAction } from 'aws-cdk-lib/custom-resources/lib/helpers-internal';
+import { ScheduleTargetBase, ScheduleTargetBaseProps } from './target';
+
+/**
+ * AWS read-only API action name prefixes that are not supported by EventBridge Scheduler.
+ *
+ * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/managing-targets-universal.html
+ */
+const NOT_SUPPORTED_ACTION_PREFIX = [
+  'get',
+  'describe',
+  'list',
+  'poll',
+  'receive',
+  'search',
+  'scan',
+  'query',
+  'select',
+  'read',
+  'lookup',
+  'discover',
+  'validate',
+  'batchGet',
+  'batchDescribe',
+  'batchRead',
+  'transactGet',
+  'adminGet',
+  'adminList',
+  'testMigration',
+  'retrieve',
+  'testConnection',
+  'translateDocument',
+  'isAuthorized',
+  'invokeModel',
+];
+
+/**
+ * Properties for a Universal Target
+ */
+export interface UniversalProps extends ScheduleTargetBaseProps {
+  /**
+   * The AWS service to call.
+   *
+   * This must be in lowercase.
+   */
+  readonly service: string;
+
+  /**
+   * The API action to call.
+   *
+   * You cannot use read-only API actions such as common GET operations.
+   *
+   * Also, This must be in camelCase.
+   */
+  readonly action: string;
+
+  /**
+   * The IAM policy statements to allow the API call.
+   *
+   * These policies will be added to the role used to invoke the API.
+   *
+   * @default - extract the permission from the API call
+   */
+  readonly policyStatements?: PolicyStatement[];
+}
+
+/**
+ * Use a wider set of AWS API as a target for AWS EventBridge Scheduler.
+ *
+ * @see https://docs.aws.amazon.com/scheduler/latest/UserGuide/managing-targets-universal.html
+ */
+export class Universal extends ScheduleTargetBase implements IScheduleTarget {
+  constructor(
+    private readonly props: UniversalProps,
+  ) {
+    const service = props.service;
+    const action = props.action;
+
+    if (!Token.isUnresolved(service) && service !== service.toLowerCase()) {
+      throw new Error(`API service must be lowercase, got: ${service}`);
+    }
+    if (!Token.isUnresolved(action) && !action.startsWith(action[0]?.toLowerCase())) {
+      throw new Error(`API action must be camelCase, got: ${action}`);
+    }
+    if (!Token.isUnresolved(action) && NOT_SUPPORTED_ACTION_PREFIX.some(prefix => action.startsWith(prefix))) {
+      throw new Error(`Read-only API action is not supported by EventBridge Scheduler: ${service}:${action}`);
+    }
+
+    const arn = `arn:${Aws.PARTITION}:scheduler:::aws-sdk:${service}:${action}`;
+    super(props, arn);
+  }
+
+  protected addTargetActionToRole(role: IRole): void {
+    // If policyStatements are not provided or are empty, add a policy statement extracted from the API call
+    if (!this.props.policyStatements?.length) {
+      role.addToPrincipalPolicy(new PolicyStatement({
+        actions: [awsSdkToIamAction(this.props.service, this.props.action)],
+        resources: ['*'],
```

Comment:
It would be good to specify the resources without specifying explicitly actions.

Reply:
Because having both `policyStatements` and `iamResources` in the props makes it hard to understand how each property relates, I added `iamResources`, `iamAction`, and `iamConditions` to make it easier to modify the primary IAM policy.
Additionally, `policyStatements` was renamed to `additionalPolicyStatements` to allow users to add custom IAM policy statements.

https://github.com/aws/aws-cdk/pull/32341/commits/e0469c67c7f532f0d7247ae951b73f569c657aee
https://github.com/aws/aws-cdk/pull/32341/commits/e211940082d75304b81a6d7d53817891ed0ce3f0
https://github.com/aws/aws-cdk/pull/32341/commits/bec041449ac2c37ae9176e59ae8a606e73ca3c95

Reply:
> Because having both `policyStatements` and `iamResources` in the props makes it hard to understand how each property relates

It is difficult to predict what IAM policy will be created when both properties are specified, for example, without actually doing it.